### PR TITLE
Create agenda list table view controller

### PR DESCRIPTION
**Pull request by MikePeralta27**
Create ContactListViewController and ContactDetailsViewController

### CODE REVIEW

File: Agenda- IOS16/Controllers/ContactDetailsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    
+    @IBOutlet weak var nameTextField: UITextField!
+    @IBOutlet weak var lasNameTextField: UITextField!
+    @IBOutlet weak var phoneNumberTextField: UITextField!
+    
+    var contacts = [Contact]()
+    var selectedContact = Contact()
+//    {
+//        willSet{
+//            //create method
```
Comment (jastinmartinez) [inline Agenda- IOS16/Controllers/ContactDetailsViewController.swift]: remove commented code

File: Agenda- IOS16/Controllers/ContactDetailsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+//        }
+//    }
+
+    let context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setData(contact: selectedContact)
+
```
Comment (jastinmartinez) [inline Agenda- IOS16/Controllers/ContactDetailsViewController.swift]: remove space of this

File: Agenda- IOS16/Controllers/ContactDetailsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    
+
+    
+    
+    //MARK: - Model Manupulation Methods
+
+//    func loadContactDetails(){
+//
+//        let request: NSFetchRequest<Contact> = Contact.fetchRequest()
+//
```
Comment (jastinmartinez) [inline Agenda- IOS16/Controllers/ContactDetailsViewController.swift]: remove all. this commented code.

File: Agenda- IOS16/Controllers/ContactDetailsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+//        do {
+//            contacts = try context.fetch(request)
+//        } catch {
+//            print("Error fetching data from context \(error)")
+//        }
+//    }
+    
+    func setData(contact: Contact){
+
```
Comment (jastinmartinez) [inline Agenda- IOS16/Controllers/ContactDetailsViewController.swift]: remove extra space over here.

File: Agenda- IOS16/Controllers/ContactListViewController.swift [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        let action = UIAlertAction(title: "Add Contact", style: .default){ (action) in
+            
+            let newContact = Contact(context: self.context)
+            newContact.name = nameTextField.text!
+            newContact.lastName = lastNameTextField.text!
+            newContact.phone = phoneTextField.text!
+            self.contacts.append(newContact)
+            
+            self.saveContact()
+
```
Comment (jastinmartinez) [inline Agenda- IOS16/Controllers/ContactListViewController.swift]: remove space

File: Agenda- IOS16/AppDelegate.swift [hunk 6]
```diff
@@ -35,26 +35,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
     lazy var persistentContainer: NSPersistentContainer = {
-        /*
-         The persistent container for the application. This implementation
-         creates and returns a container, having loaded the store for the
-         application to it. This property is optional since there are legitimate
-         error conditions that could cause the creation of the store to fail.
-        */
-        let container = NSPersistentContainer(name: "Agenda__IOS16")
+        
+        let container = NSPersistentContainer(name: "DataModel")
```
Comment (jastinmartinez) [inline Agenda- IOS16/AppDelegate.swift]: ad an extension of this appDelegate with an enum call PersistenceKey

and should be as the following 

`enum PersistanceKey: String  {
  case Key = "DataModel"
}
`

you can used it as the following 


`let container = NSPersistentContainer(name: PersistanceKey.key.rawValue)`

**Review by jastinmartinez — CHANGES_REQUESTED** [review]

File: Agenda- IOS16/AppDelegate.swift (hunk 6 above)
Comment (MikePeralta27) [reply Agenda- IOS16/AppDelegate.swift]: Addressed

File: Agenda- IOS16/Controllers/ContactDetailsViewController.swift (hunk 1 above)
Comment (MikePeralta27) [reply Agenda- IOS16/Controllers/ContactDetailsViewController.swift]: Commented code removed

File: Agenda- IOS16/Controllers/ContactDetailsViewController.swift (hunk 2 above)
Comment (MikePeralta27) [reply Agenda- IOS16/Controllers/ContactDetailsViewController.swift]: Space removed

File: Agenda- IOS16/Controllers/ContactDetailsViewController.swift (hunk 4 above)
Comment (MikePeralta27) [reply Agenda- IOS16/Controllers/ContactDetailsViewController.swift]: Extra space removed

File: Agenda- IOS16/AppDelegate.swift (hunk 6 above)
Comment (MikePeralta27) [reply Agenda- IOS16/AppDelegate.swift]: AppDelegate extension added.

File: Agenda- IOS16/Controllers/ContactListViewController.swift (hunk 5 above)
Comment (MikePeralta27) [reply Agenda- IOS16/Controllers/ContactListViewController.swift]: Space removed.

File: Agenda- IOS16/Controllers/ContactDetailsViewController.swift (hunk 3 above)
Comment (MikePeralta27) [reply Agenda- IOS16/Controllers/ContactDetailsViewController.swift]: Commented code removed.

**Review by jastinmartinez — APPROVED** [review]

**Pull request merged**